src/ui/common/src/components/integrations/dialogs/dialog.tsx

```diff
@@ -482,23 +482,24 @@ export function isConfigComplete(
   service: Service
 ): boolean {
   switch (service) {
-    case 'S3':
-      return isS3ConfigComplete(config as S3Config);
     case 'Athena':
       return isAthenaConfigComplete(config as AthenaConfig);
-    case 'Kubernetes':
-      return isK8sConfigComplete(config as KubernetesConfig);
     case 'Conda':
       return true;
     case 'Databricks':
       return isDatabricksConfigComplete(config as DatabricksConfig);
     case 'Email':
       return isEmailConfigComplete(config as EmailConfig);
+    case 'Kubernetes':
+      return isK8sConfigComplete(config as KubernetesConfig);
+    case 'S3':
+      return isS3ConfigComplete(config as S3Config);
     case 'Slack':
       return isSlackConfigComplete(config as SlackConfig);
     case 'Spark':
       return isSparkConfigComplete(config as SparkConfig);
-
+    case 'Snowflake':
```

Comment:
As we keep adding integrations, this switch statement is going to keep growing. This actually seems like a great use case for a TypeScript class. 

I know it's a bit different than the other code that we have in our frontend codebase, but it would be nice to have a base integration class that each of these Config objects implement and provide their own 'isConfigComplete' function implementations.

This doesn't have to happen now of course, but just some food for thought when we get to hundreds of integrations :)

Reply:
@agiron123 that's a good idea, could you create a linear task for this?

Reply:
Sure thing @likawind, you can find it here:
https://linear.app/aqueducthq/issue/ENG-2556/integrations-page-refactor